CONTRIBUTING.md

```diff
@@ -1,35 +1,77 @@
-
-How to Contribute Code to ROOT
-==============================
+# How to Contribute Code to ROOT
 
 Thank you for your interest in contributing to ROOT!  We strongly welcome and appreciate such contributions!
 This short guide tries to make contributing as quick and painless as possible.
 
-Your Pull Request
------------------------
+## Your Code Contribution
 
 The source code for ROOT is kept in [GitHub](https://github.com/root-project/root).
 Changes go through pull requests ("PRs").
 The primary branch for development is `master`.
-Visit [this page](https://root.cern/for_developers/creating_pr) for the mechanics on how to
-create pull requests.
+
+> [!IMPORTANT]
+> We require PRs to cleanly apply to master without a merge commit, i.e. through "fast-forward".
+> Please follow the [coding conventions](https://root.cern.ch/coding-conventions), as this is a simple item for
+> reviewers to otherwise get stuck on.
+> To make your (and our own) life easier, we provide a
+> [`clang-format` configuration file](https://github.com/root-project/root/blob/master/.clang-format).
 
 By providing code, you agree to transfer your copyright on the code to the "ROOT project".
 Of course you will be duly credited: for sizable contributions your name will appear in the
-[CREDITS](https://raw.githubusercontent.com/root-project/root/master/README/CREDITS){:target="_blank"}
+[CREDITS](https://raw.githubusercontent.com/root-project/root/master/README/CREDITS)
 file shipped with every binary and source distribution.
 The copyright transfer helps us with effectively defending the project in case of litigation.
 
-:warning: We require PRs to cleanly apply to master without a merge commit, i.e. through "fast-forward".
-Please follow the [coding conventions](https://root.cern.ch/coding-conventions),
-as this is a simple item for reviewers to otherwise get stuck on.
+## Your Commit
+
+Each commit is a self-contained, _atomic_ change. This means that:
+1. **Each commit should be able to successfully build ROOT.**
+Doing so makes traveling through the git history, for example during a `git bisect` much easier.
+Ideally, the commit also should not depend on other commits to _run_ ROOT.
+2. **Each commit does not contain more than one independent change.**
+This allows us to revert changes when needed, without affecting anything else.
+
+> [!TIP]
+> During a code review, it may be useful to make smaller commits to track intermediate changes, and rebase after the PR
+> is approved to ensure the above points are met and to reduce clutter.
+
+### Your Commit Message
+
+The commit summary (i.e. the first line of the commit message) should be preceded by the part of ROOT that is affected
```

Comment:
We should be ready to list the parts of ROOT and write that if the collaborator does find anything appropriate, we are here to help.

Reply:
Do we want to add this list to this document, or only keep it internally?